### PR TITLE
add -a flag which outputs assembly of ICs

### PR DESCRIPTION
**Pull request by tjhance**
In an effort to better understand the performance of our ICs

Add `-a` flag, which will dump a disassembly of all the ICs to stderr. Next it would be nice to construct a good test program that exercises a representative sample of ICs.

Here's an example output:

```
    nop   
; _getAttr
    movq    (%rdi), %rax
; _addAttrGuard
    movabsq $79188951816, %r8
    cmpq    336(%rax), %r8
    jne 613
; _addAttrGuard
    movabsq $79188493320, %r8
    cmpq    (%rax), %r8
    jne 594
; _addAttrGuard
    movabsq $79188828936, %r8
    cmpq    384(%rax), %r8
    jne 571
; _getAttr
    movq    392(%rax), %rax
; _getAttr
    movq    32(%rax), %rax
; _addAttrGuard
    movabsq $79188558856, %r8
    cmpq    (%rax), %r8
    jne 541
; _getAttr
    movq    (%rdi), %r8
; _addGuard
    movabsq $79188940584, %r8
    cmpq    %rax, %r8
    jne 519
; _call 
    movabsq $1311768467294899695, %rax
    incl    (%rax)
    movabsq $10410624, %r11
    callq   *%r11 
; commitReturning
; mark inside ic
    movabsq $1311768467294899695, %rcx
    decl    (%rcx)
; live outs
```

### CODE REVIEW

**Comment by undingen** [comment]
nice, this will help debugging the new jit tier

**Comment by tjhance** [comment]
One of the llvm headers I had to include required me to add `-Wno-non-virtual-dtor` to our build. That kinda sucks...

**Comment by toshok** [comment]
oh that output is awesome

**Comment by tjhance** [comment]
I added a script that runs all the `*_ics.py` tests and outputs some stats

File: CMakeLists.txt [hunk 1]
```diff
@@ -178,7 +178,7 @@ if(ENABLE_OPROFILE)
 endif()
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Werror -Wreturn-type -Wno-sign-compare -Wno-unused -Wno-unused-parameter -fno-omit-frame-pointer")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_C_FLAGS} -std=c++11 -fno-rtti -fexceptions -fvisibility-inlines-hidden -ffunction-sections -fdata-sections -Woverloaded-virtual -Wno-invalid-offsetof -Wcast-qual -Wno-sign-conversion -Wnon-virtual-dtor -Winit-self -Wmissing-include-dirs -Wstrict-overflow=5 -Wpointer-arith -Wtype-limits -Wwrite-strings -Wempty-body -Waggregate-return -Wmissing-field-initializers -Wredundant-decls -Winline -Wint-to-pointer-cast -Wlong-long -Wvla -Wno-attributes -g")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_C_FLAGS} -std=c++11 -fno-rtti -fexceptions -fvisibility-inlines-hidden -ffunction-sections -fdata-sections -Woverloaded-virtual -Wno-invalid-offsetof -Wcast-qual -Wno-sign-conversion -Wno-non-virtual-dtor -Winit-self -Wmissing-include-dirs -Wstrict-overflow=5 -Wpointer-arith -Wtype-limits -Wwrite-strings -Wempty-body -Waggregate-return -Wmissing-field-initializers -Wredundant-decls -Winline -Wint-to-pointer-cast -Wlong-long -Wvla -Wno-attributes -g")
```
Comment (kmod) [inline CMakeLists.txt]: It'd be nice to not turn this off -- maybe we could do a `#pragma clang diagnostic ignored "-Wnon-virtual-dtor"` + push+pop around the offending include?  [Here's](https://github.com/dropbox/pyston/blob/master/from_cpython/Modules/posixmodule.c#L5902) an example of another place we did that.  Not sure if we'd need another pragma for gcc.

**Comment by tjhance** [comment]
addressed your comment, kmod

**Comment by tjhance** [comment]
ping

is this okay to be merged in?

**Comment by kmod** [comment]
Ah whoops.  I think it has the issue that it shows the assembly pre-relocation, but I think your next PR addresses that.

**Pull request merged**